smart_home/src/devices/ac_unit.cc

```diff
@@ -3,13 +3,23 @@
 
 namespace devices {
 
-void AcUnit::SetFanSpeed(int value) {
+void AcUnit::SetIsOn(bool value) {
+    if (is_on != value) {
+        std::cout << "Turning AC unit " << (value ? "on" : "off") << "..." << std::endl;
+        is_on = value;
+    }
+    else {
+        std::cout << "AC unit already turned " << (value ? "on" : "off") << "." << std::endl;
+    }
+}
+
+void AcUnit::SetFanSpeed(unsigned char value) {
     if (fan_speed != value) {
-        std::cout << "Setting fan speed to " << value << "%." << std::endl;
+        std::cout << "Setting fan speed to " << (int)value << "%." << std::endl;
```

Comment:
2 issues here:
- C-Style Casting (worst possible way to cast in C++ due to not having control over it, C-Style cast in modern C++ tries static_cast<> first, dynamic_cast<> afterwards and if neither succeeds, goes to reinterpret_cast<> where anything can happen)
- Needing to cast shows that you have a design flaw in your code :). In this case, using unsigned char. Discussed a bit more about it in a previous comment